File: 백준/Silver/20922. 겹치는 건 싫어/겹치는 건 싫어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Sol {

    public int solution() throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]);
        int K = Integer.parseInt(input[1]);

        input = br.readLine().split(" ");
        int[] numbers = new int[N];
        for (int i = 0; i < N; i++) {
            numbers[i] = Integer.parseInt(input[i]);
        }

        int maxLength = 0, curLength = 0;
        int[] cnt = new int[100_001];
        int p1 = 0;
        for (int p2 = 0; p2 < N; p2++) {
            curLength++;
            int right = numbers[p2];
            cnt[right]++;
            while (cnt[right] > K) {
                int left = numbers[p1++];
                cnt[left] -= 1;
                curLength--;
            }

            if (maxLength < curLength) {
                maxLength = curLength;
            }
        }

        return maxLength;
    }
}

public class Main {

    public static void main(String[] args) throws Exception {
        Sol sol = new Sol();
        System.out.println(sol.solution());
    }
}

File: 백준/Silver/20922. 겹치는 건 싫어/겹치는 건 싫어.py
N, K = map(int, input().split())
arr = [
    *map(int, input().split())
]

cnt = [0] * 100_001

p1 = 0
for i in range(K):
    cnt[arr[i]] += 1

answer = K
for p2 in range(K, N):
    cnt[arr[p2]] += 1
    while cnt[arr[p2]] > K:
        cnt[arr[p1]] -= 1
        p1 += 1
    answer = max(answer, p2-p1+1)
print(answer)
